### PR TITLE
Various Fixes to ASV

### PR DESCRIPTION
**Pull request by amzn-phist**
- Replaced deprecated CMake PAL functions with current version (gets rid of cmake warnings).
- Fixed uses of Uuid based on recent changes in o3de.
- Fixed spelling of a type.

### CODE REVIEW

**Review by rgba16f — APPROVED** [review]

**Comment by rgba16f** [comment]
Please do a run on ASV's commit wizard and validate nothing broke on DX12 and vulkan

File: Gem/Code/Source/BakedShaderVariantExampleComponent.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ namespace AtomSampleViewer
 
         Data::AssetId modelAssetId;
         Data::AssetCatalogRequestBus::BroadcastResult(
-            modelAssetId, &Data::AssetCatalogRequestBus::Events::GetAssetIdByPath, Products::ModelFilePath, nullptr, false);
+            modelAssetId, &Data::AssetCatalogRequestBus::Events::GetAssetIdByPath, Products::ModelFilePath, AZ::Uuid::CreateNull(), false);
```
Comment (lumberyard-employee-dm) [inline Gem/Code/Source/BakedShaderVariantExampleComponent.cpp:78]: FYI, brace initialization can be used here `{}` .
The AZ::Uuid class now default constructs to a Null Uuid.

**Review by lumberyard-employee-dm — APPROVED** [review]
Thanks for fixing this.
I was building the SDK layout to build AtomSampleViewer against.

File: Gem/Code/Source/BakedShaderVariantExampleComponent.cpp (hunk 1 above)
Comment (rgba16f) [reply Gem/Code/Source/BakedShaderVariantExampleComponent.cpp:78]: Personally I prefer CreateNull vs {}. With brace initialization I need to go look up what UUID does in that case, CreateNull tells me explicitly.

Comment (amzn-phist) [reply Gem/Code/Source/BakedShaderVariantExampleComponent.cpp:78]: If you prefer that, I can do it.  Saying `CreateNull()` seems to be a bit more "explicit" telling users what's being passed.

Comment (lumberyard-employee-dm) [reply Gem/Code/Source/BakedShaderVariantExampleComponent.cpp:78]: It doesn't matter to be me how one does it.
But all these initializations are now equivalent
```
{}
AZ::Uuid()
AZ::Uuid{}
AZ::CreateNull()
```

**Comment by amzn-phist** [comment]
dx12
```
Asserts: 0 
Errors: 0 
Warnings: 0 
Screenshot errors: 0 
Screenshot warnings: 0 
```

vulkan
```
Asserts: 0 
Errors: 0 
Warnings: 0 
Screenshot errors: 0 
Screenshot warnings: 0 
```

**Pull request merged**